codice/ddf
# DDF-3434 Sanitizes preferences object in Intrigue UI.

#### What does this PR do?

Sanitizes preferences object in Intrigue UI.

#### Who is reviewing it? 

@garrettfreibott 
@tbatie 

#### Select relevant component teams:

@codice/ui 
@codice/security 

#### Choose 2 committers to review/merge the PR. 

@andrewkfiedler
@stustison

#### How should this be tested?

- Full build of `catalog/ui/catalog-ui-search`.
- Ensure that layouts are saved correctly
  - Modify the layout
  - Ensure it persists across page reloads
- Ensure any html in the golden layout preferences object is sanitized

#### What are the relevant tickets?

[DDF-3434](https://codice.atlassian.net/browse/DDF-3434)

#### Checklist:

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
